app/controllers/profile_orders_controller.rb

```diff
@@ -5,9 +5,10 @@ def index
   end
 
   def show
-
     @order = Order.find(params[:id])
-
+    if @order.status == "cancelled" || @order.status == "shipped"
+    else
+      @order.update(status: @order.find_order_status(params[:id]))
```

Comment:
Should probably refactor but Eric and I got this to work so in the wise words of some redneck "if it works don't fix it"